source/Octopus.Cli/Commands/UserProfileHelper.cs

```diff
@@ -0,0 +1,60 @@
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Octopus.Cli.Commands
+{
+    public static class UserProfileHelper
```

Comment:
Not sure if this is worth it... but you could implement each supported shell in a different class, and then use DI to inject them all - that way adding support for a new shell is as simple as adding a new file, rather than modifying in a bunch of places?